mlflow/tracking/client.py

```diff
@@ -57,8 +57,8 @@ def create_run(self, experiment_id, user_id=None, run_name=None, source_type=Non
             experiment_id=experiment_id,
             user_id=user_id if user_id is not None else _get_user_id(),
             run_name=run_name,
-            source_type=source_type,
-            source_name=source_name,
+            source_type=source_type if source_type is not None else SourceType.LOCAL,
```

Comment:
Added sane defaults here to look closer to the Java API; otherwise, `create_run(experiment_id)` does not work. Found via the integration test!